packages/components/src/chart/d3chart/chart.js

```diff
@@ -89,6 +89,24 @@ class D3Chart extends Component {
 		type === 'bar' && drawBars( g, data, adjParams );
 	}
 
+	shouldBeCompact() {
+		const {	data, margin, type, width } = this.props;
+		const widthWithoutMargins = width - margin.left - margin.right;
+		const columnsPerDate = data && data.length ? Object.keys( data[ 0 ] ).length - 1 : 0;
+		const minimumWideWidth = data.length * ( columnsPerDate + 1 );
+
+		return type === 'bar' && widthWithoutMargins < minimumWideWidth;
+	}
+
+	getWidth() {
+		const {	data, margin, type, width } = this.props;
+		const widthWithoutMargins = width - margin.left - margin.right;
+		const columnsPerDate = data && data.length ? Object.keys( data[ 0 ] ).length - 1 : 0;
+		const minimumWidth = this.shouldBeCompact() ? data.length * columnsPerDate : data.length * ( columnsPerDate + 1 );
+
+		return type === 'bar' ? Math.max( widthWithoutMargins, minimumWidth ) : widthWithoutMargins;
```

Comment:
Since these functions get called so much, could it be worth it to exit early with `widthWithoutMargins` for bar charts. Same above with `shouldBeCompact`.

Reply:
Good idea! Implemented here: e782db0147d9b14cd29bc1977164c37b6060b7c6.